### PR TITLE
[Validation] Add User registration validator

### PR DESCRIPTION
**Pull request by makrushinoff**
From issue #18

### CODE REVIEW

File: src/test/java/ua/com/sportfood/services/CustomerValidationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        Customer result = null;
+        try{
+            result = testInstance.checkCustomerIsPresent(form);
+        } catch (PresentCustomerException exc) {
+
+        }
+
+        assertEquals(testCustomer, result);
+    }
+}
```
Comment (MLGMAG) [inline src/test/java/ua/com/sportfood/services/CustomerValidationServiceTest.java:106]: Add new line at the end of file

File: src/test/java/ua/com/sportfood/services/CustomerValidationServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            "last1",
+            "num1"),
+            new Customer(
+                    "email2",
+                    "user2",
+                    "pass2",
+                    "first2",
+                    "last2",
+                    "num2"
+            ));
```
Comment (MLGMAG) [inline src/test/java/ua/com/sportfood/services/CustomerValidationServiceTest.java:41]: extract this values to constants

File: src/test/java/ua/com/sportfood/services/CustomerValidationServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class CustomerValidationServiceTest {
+
+    @InjectMocks
+    private CustomerValidationService testInstance;
+
+    @Mock
+    private CustomerDAO customerDAO;
```
Comment (MLGMAG) [inline src/test/java/ua/com/sportfood/services/CustomerValidationServiceTest.java:25]: Mocks at top, test instance at bottom

File: src/main/java/ua/com/sportfood/services/CustomerValidationService.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+     * @param form - data from customer, that is got from registration form
+     * @return new Customer object.
+     * @see Customer
+     */
+    public Customer checkCustomerIsPresent(RegistrationForm form) throws PresentCustomerException {
+        List<Customer> allUsers = customerDAO.findAll();
+        boolean containsCustomer = allUsers.stream().anyMatch(customer ->
+                        customer.getUsername().equals(form.getUsername()) ||
+                        customer.getPhoneNumber().equals(form.getPhoneNumber()) ||
+                        customer.getEmail().equals(form.getEmail()));
```
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/services/CustomerValidationService.java:41]: This if statement is incomprehensible, can you extract it into separate method?

File: src/main/java/ua/com/sportfood/exceptions/PresentCustomerException.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package ua.com.sportfood.exceptions;
+
+public class PresentCustomerException extends Exception{
```
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/exceptions/PresentCustomerException.java:3]: Format code pls (space after Exception)

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: src/main/java/ua/com/sportfood/exceptions/PresentCustomerException.java (hunk 5 above)
Comment (makrushinoff) [reply src/main/java/ua/com/sportfood/exceptions/PresentCustomerException.java:3]: Done

File: src/main/java/ua/com/sportfood/services/CustomerValidationService.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+import ua.com.sportfood.models.Customer;
+
+import java.util.List;
+
+/**
+ * CustomerValidator is a class, that should validate input data
+ * from {@link RegistrationForm}
+ */
+@Service
+public class CustomerValidationService {
```
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/services/CustomerValidationService.java:16]: Change class name to RegistrationFormValidator

File: src/main/java/ua/com/sportfood/services/CustomerValidationService.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    private final CustomerDAO customerDAO;
+    private static final String PRESENT_VALUE_EXCEPTION = "Customer is already present in database. Field: '%s', value: '%s';";
```
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/services/CustomerValidationService.java:18]: Insert new line

File: src/main/java/ua/com/sportfood/services/CustomerValidationService.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+            throw new PresentCustomerException(PHONE_NUMBER_FIELD, form.getPhoneNumber());
+        }
+    }
+
+    private void usernameCheck(RegistrationForm form, Customer customer) throws PresentCustomerException {
+        boolean usernameCheck = customer.getUsername().equals(form.getUsername());
+        if (usernameCheck) {
+            throw new PresentCustomerException(USERNAME_FIELD, form.getUsername());
+        }
+    }
```
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/services/CustomerValidationService.java:75]: Rename methods to *validation
Example emailCheck -> emailValidation

File: src/main/java/ua/com/sportfood/exceptions/PresentCustomerException.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
+package ua.com.sportfood.exceptions;
+
+public class PresentCustomerException extends RuntimeException {
```
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/exceptions/PresentCustomerException.java:3]: Rename class into RegistrationFormValidationException

File: src/main/java/ua/com/sportfood/services/CustomerValidationService.java (hunk 6 above)
Comment (MLGMAG) [inline src/main/java/ua/com/sportfood/services/CustomerValidationService.java:16]: Create interface -> Validator with single method validate()

**Review by MLGMAG — CHANGES_REQUESTED** [review]

**Review by MLGMAG — APPROVED** [review]

**Pull request merged**